R/get_ref.R

```diff
@@ -51,8 +51,19 @@ get_ref_min_incl_cran.remote_ref_github <- function(remote_ref, op = "", op_ver
 #' Check if package is available on CRAN.
 #' @importFrom pkgcache meta_cache_list
 #' @keywords internal
-check_if_on_cran <- function(remote_ref) {
-  nrow(pkgcache::meta_cache_list(remote_ref$package)) > 0
+#'
+#' @examplesIf Sys.getenv("R_USER_CACHE_DIR", "") != ""
+#' verdepcheck:::check_if_on_cran(list(package = "dplyr"))
+#' verdepcheck:::check_if_on_cran(list(package = "dplyr"), list(op = ">=", op_ver = "1.1.0"))
+#' verdepcheck:::check_if_on_cran(list(package = "dplyr"), list(op = ">=", op_ver = "9999.9.99"))
+#' verdepcheck:::check_if_on_cran(list(package = "dplyr"), list(op = "<", op_ver = "0.0.0"))
+check_if_on_cran <- function(remote_ref, version = NULL) {
+  cran_listings <- pkgcache::meta_cache_list(remote_ref$package)
+  if (is.null(version) || NROW(cran_listings) == 0) {
+    return(NROW(cran_listings) > 0)
+  }
```

Comment:
condition checked twice

```suggestion
  if (is.null(version)) {
    return(NROW(cran_listings) > 0)
  }
```